File: rumahsehat/src/main/java/TA_A_ME_61/RumahSehat/model/TagihanModel.java
package TA_A_ME_61.RumahSehat.model;

import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "tagihan")
public class TagihanModel implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "kode")
    private String kode;

    @NotNull
    @Column(name = "tanggalTerbuat", nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime tanggalTerbuat;

    @Column(name = "tanggalBayar")
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime tanggalBayar;

    @NotNull
    @Column(name = "isPaid", nullable = false)
    private Boolean isPaid;

    @NotNull
    @Column(name = "jumlahTagihan", nullable = false)
    private Long jumlahTagihan;

    @OneToOne(cascade = CascadeType.ALL) 
    private AppointmentModel appointment;

}

File: rumahsehat_app/lib/models/tagihan_model.dart
class TagihanModel {
  final String kode;
  final String tanggalTerbuat;
  final String tanggalBayar;
  final bool isPaid;
  final int jumlahTagihan;

  const TagihanModel({
    required this.kode,
    required this.tanggalTerbuat,
    required this.tanggalBayar,
    required this.isPaid,
    required this.jumlahTagihan,
  });

  factory TagihanModel.fromJson(Map<String, dynamic> json) {
    return TagihanModel(
      kode: json['kode'],
      tanggalTerbuat: json['tanggalTerbuat'],
      tanggalBayar: json['tanggalBayar'] ?? "Belum dibayar",
      isPaid: json['isPaid'],
      jumlahTagihan: json['jumlahTagihan'],
    );
  }
}
